salt/states/user.py

```diff
@@ -373,6 +373,14 @@ def present(name,
            'result': True,
            'comment': 'User {0} is present and up to date'.format(name)}
 
+    # the comma is used to separate field in GECOS, thus resulting into
+    # salt adding the end of fullname each time this function is called
+    for gecos_field in ['fullname', 'roomnumber', 'workphone', 'homephone']:
+        if isinstance(locals()[gecos_field], type('')) and ',' in locals()[gecos_field]:
```

Comment:
This will work but in the future, you can always use `salt.ext.six.string_types` to ensure forward compabability with string type checking.